instance/models/server.py

```diff
@@ -145,7 +150,8 @@ def _set_status(self, status):
 
         self.status = status
         self.logger.info('Changed status: %s', self.status)
-        self.save()
+        if commit:
+            self.save()
```

Comment:
Why not keeping saving the status every time it changes? Is it to save writes in the DB? Having it always saved allows to reliably trigger events on status change.